src/core/meshcop/commissioner.cpp

```diff
@@ -164,27 +164,40 @@ otError Commissioner::Start(otCommissionerStateCallback  aStateCallback,
     SetState(OT_COMMISSIONER_STATE_PETITION);
 
 exit:
+    if (error != OT_ERROR_NONE)
+    {
+        Get<Coap::CoapSecure>().Stop();
+    }
     return error;
 }
 
-otError Commissioner::Stop(void)
+otError Commissioner::Stop(bool aResign)
```

Comment:
is it necessary to have the `bool aResign` parameter?
shouldn't the resign only be required when stopping an active commissioner?

if the commissioner is stopped before receiving the PET.rsp with accept status, the commissioner itself won't enter active mode, and the leader will expire commissioner session automatically after keep-alive timeout.